dFusion/dFusion.rst

```diff
@@ -184,170 +182,230 @@ TODO
+
+
+=================  =================  =====  ================= 
+                   Index_1            ...    Index_S
+=================  =================  =====  =================
+(price, index)     P_1, token_index   ...    P_S, token_index
+=================  =================  =====  =================
+
+of all prices relative to a reference token **Token_1** with token_index = 0.
+The maximal size of the array is the total amount of tokens registed in the anchor contract.
+Though it is expected that only a small fraction of tokens is actually touched and has a positive trading volume.
+All tokens with a positive trading volume are required to be listed in the vector and have a well defined price.
 
-After the solution submission period, the best solution with the highest trading surplus will be chosen by the anchor contract. 
-The submitter of this solution then needs to post the full solution into the ethereum chain as calldata payload. 
-The solution is a new stateHash with the updated account balances, a price vector **P**:
+Since prices are arbitrage-free, we can calculate the **price Token_i: Token_k** =  **(Token_i:Token_1):(Token_1:Token_k)**.
+Each price is a 32-bit number and each token_index is a 10-bit number.
 
+The touched orders is a vector of the following format:
 
-=====  =================  =====  ================= 
- P      Token_1:Token_1    ...    Token_T:Token_1 
-=====  =================  =====  =================
-price   p_1                ...    p_T
-=====  =================  =====  =================
+=================  =================  =====  ================= 
+                   Index_1            ...      Index_K
+=================  =================  =====  =================
+orders             orderIndex_1       ...    orderIndex_K
+=================  =================  =====  =================
 
-of all prices relative to a reference token **Token_1**. Since prices are arbitrage-free, we can calculate the **price Token_i: Token_k** =  **(Token_i:Token_1):(Token_1:Token_k)**.
+The orderIndex is always referring to the orderIndex in the orderstream.
+Since during the batch closing a snap-shot of the latest order batch is taken, we can refer to any order in the stream relative to that snap-shot.
+The orderIndex 2**24 will reference to the last order in the batch.
+And the orderIndex 0 will refere to the order submitted 2**24 order before the last one.
+Each orderIndex is a 24 bit number.
 
-Along with the prices, the solution submitter also has to post a vector **V** of **buyVolumes** for each order:
+The size of orders in the batch is bound by the amount of orders that we can verify on-chain within one fraud-proof.
+Currently, the limit should be 1024 orders.
+Solutons with any K< 2**10 can be valid solutions and maximizing the traders utility.
 
+Along with the orders, the solution submitter also has to post a vector **V** of **buyVolumes** and **sellVolumes** for each order:
```

Comment:
I believe we should be using `Amounts` here rather than `Volumes`. This is due to the fact that one of them is in reference to the unit amount of execution where the other (volume) is with respect to price. `volume = amount * price`